### PR TITLE
Refactor gatsby node.js file

### PR DESCRIPTION
**Pull request by mitra-deriv**
Changes:

-  Revise the gatsby-node.js file to streamline the management of URL redirections. Move all redirection URLs to a redirects.json file, in line with best practices outlined in the Gatsby documentation for handling redirects and rewrites.
Doc:
https://www.gatsbyjs.com/docs/how-to/cloud/working-with-redirects-and-rewrites/

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: redirects.json [hunk 1]
```diff
@@ -0,0 +1,117 @@
+[
+    {
+        "fromPath": "/careers/people-management",
+        "toPath": "/careers/human-resources/",
+        "status": 301
```
Comment (mitra-deriv) [inline redirects.json:5]: Do I need to use status here or as we have it here https://github.com/binary-com/deriv-com/pull/5533/files#diff-a6f3904c41211be170d9b0eb87e49ea8d02d0fd5a7224d6772174b425dacf90fR14 in gatsby-node.js file is sufficient ?

File: gatsby-node.js [hunk 2]
```diff
@@ -1,194 +1,26 @@
...
-            isPermanent: true,
-        })
-        createRedirect({
-            fromPath: `/derivez`,
-            toPath: `/deriv-ez/`,
-            redirectInBrowser: true,
-            isPermanent: true,
-        })
-    }
-    const is_english = process.env.GATSBY_LANGUAGE === 'en'
```
Comment (habib-deriv) [inline gatsby-node.js:189]: This is needed for vercel english pages

File: gatsby-node.js [hunk 3]
```diff
@@ -1,194 +1,26 @@
...
+    })
+}
 
-    Object.keys(is_english ? language_config_en : language_config).map((lang) => {
```
Comment (habib-deriv) [inline gatsby-node.js:191]: This is needed too

File: gatsby-node.js [hunk 4]
```diff
@@ -1,194 +1,26 @@
 /* eslint-disable import/order */
 const language_config = require(`./i18n-config.js`)
-const language_config_en = require(`./i18n-config-en.js`)
```
Comment (habib-deriv) [inline gatsby-node.js:3]: This is needed

File: redirects.json (hunk 1 above)
Comment (habib-deriv) [reply redirects.json:5]: Yes, not needed here

File: gatsby-node.js [hunk 5]
```diff
@@ -1,194 +1,26 @@
...
 
-    if (is_p2p) {
+    redirects.forEach((redirect) => {
         createRedirect({
-            fromPath: `/p2p/v1`,
-            toPath: `/p2p`,
-            redirectInBrowser: true,
+            fromPath: redirect.fromPath,
+            toPath: redirect.toPath,
             isPermanent: true,
```
Comment (habib-deriv) [inline gatsby-node.js:14]: If you want to keep 301 status in the json file can try this
```suggestion
            isPermanent: redirect.status === 301 ? true : false
```

File: redirects.json (hunk 1 above)
Comment (habib-deriv) [reply redirects.json:5]: https://github.com/binary-com/deriv-com/pull/5533/files#r1328571235

**Review by habib-deriv — APPROVED** [review]

**Pull request closed without merging**